# Chapter 3 Hydraulics: Design of Culverts

1. [Introduction: Culvert Design](#1.-Introduction)
2. [Simulation: Inlet and Outlet Control](#2.-Simulation)
3. [Self-Assessment](#3.-Self-Assessment)

## 1. Introduction

###  What Is a Culvert?

A **culvert** is a hydraulic structure that allows water to flow under a road, railroad, embankment, or similar obstruction. Understanding the culvert’s capacity under different control conditions is critical for proper drainage, flood mitigation, and road safety.

### 📚 Functions of Culverts

- **Drainage Conveyance**: Transfers water across roads or embankments
- **Flood Control**: Prevents overtopping and erosion during storm events
- **Habitat Connectivity**: Maintains aquatic organism passage and stream continuity
- **Roadway Protection**: Prevents water accumulation and structural damage
- **Stormwater Management**: Integrates with detention and conveyance systems

---

### 🧱 Types of Culverts

| Type                  | Description                                                                 |
|-----------------------|-----------------------------------------------------------------------------|
| **Pipe Culvert**       | Circular or elliptical pipe; commonly made of concrete, steel, or HDPE     |
| **Box Culvert**        | Rectangular concrete structure; suitable for high flows and shallow cover  |
| **Arch Culvert**       | Open-bottom arch; preserves streambed and supports ecological flow         |
| **Slab Culvert**       | Flat slab over open channel; used for short spans                          |
| **Bridge Culvert**     | Hybrid between culvert and small bridge; spans wider channels              |

---

### 🧮 Classification Table: Culverts by Shape, Material, and Flow Type

| Classification Category | Subtypes / Examples                          | Notes                                                             |
|--------------------------|---------------------------------------------|-------------------------------------------------------------------|
| **Shape**                | Circular, Elliptical, Box, Arch, Slab       | Influences hydraulic capacity and structural design               |
| **Material**             | Concrete, Corrugated Steel, HDPE, PVC       | Affects durability, cost, and installation method                 |
| **Flow Type**            | Inlet-controlled, Outlet-controlled         | Determines headwater elevation and flow regime                    |
| **Installation**         | Embedded, Open-bottom, Closed-bottom        | Impacts ecological connectivity and scour potential               |
| **Hydraulic Design**     | Subcritical, Supercritical, Pressurized     | Guides sizing and energy dissipation strategies                   |

---

## ⚠️ Challenges in Culvert Design

| Challenge               | Description                                                                 |
|-------------------------|-----------------------------------------------------------------------------|
| **Hydraulic Capacity**  | Must convey design flow without overtopping or excessive headwater          |
| **Sediment Transport**  | Risk of blockage or scour if not properly sized or aligned                  |
| **Ecological Impact**   | Can disrupt fish passage and stream continuity if poorly designed           |
| **Structural Integrity**| Must withstand loads from traffic and soil pressure                         |
| **Maintenance Access**  | Requires inspection and debris removal to prevent clogging                  |
| **Energy Dissipation**  | High velocities may require aprons, baffles, or stilling basins             |

---

### 🚧 Culvert Hydraulics Classification – Ven Te Chow

{cite}`chow1959open` classified culvert flow based on the **location of hydraulic control** and the **flow regime**. These classifications help engineers determine headwater elevation, flow capacity, and whether the culvert is operating under inlet or outlet control.

---

## 📊 Culvert Flow Classification Table

| Flow Type         | Control Location | Description                                                                 |
|-------------------|------------------|-----------------------------------------------------------------------------|
| **Inlet Control** | At the entrance  | Flow is governed by the culvert entrance geometry and headwater elevation. |
| **Outlet Control**| At the exit      | Flow is governed by downstream conditions and full barrel hydraulics.      |

---

### 🔍 Subtypes of Culvert Flow (Outlet-Controlled)

| Flow Regime       | Description                                                                 |
|-------------------|-----------------------------------------------------------------------------|
| **Subcritical Flow** | Flow is slow and deep; controlled by downstream tailwater.               |
| **Supercritical Flow**| Flow is fast and shallow; controlled by upstream conditions.            |
| **Pressurized Flow** | Culvert flows full; behaves like a pipe under pressure.                  |
| **Free Surface Flow**| Flow has a free surface; behaves like open channel flow.                 |

---

### 📐 Design Implications

- **Inlet Control**: Use entrance geometry and headwater to size culvert.
- **Outlet Control**: Consider barrel roughness, slope, and tailwater elevation.
- **Critical Depth**: Helps determine control section and flow regime.
- **Energy Losses**: Include entrance, friction, and exit losses in design.

---

### 🧠 Conceptual Insight

> Ven Te Chow’s classification helps engineers identify the **governing control point**  
> and apply appropriate equations for **culvert sizing**, **headwater prediction**, and **flow regime analysis**.

> Culverts are more than just pipes under roads —  
> they are critical hydraulic structures that balance **engineering**, **environmental**, and **hydrologic** needs.

### References

{cite}`gupta2017hydrology` focuses on simplified culvert design using empirical formulas and flow classifications, emphasizing inlet/outlet control and headwater analysis. Ideal for undergraduate learners and competitive exam preparation with clear, formula-based steps.
{cite}`norman2001culvert` provides a comprehensive description integrating HEC-5, HEC-10, and HEC-13, covering hydrology, storage routing, inlet/outlet control, and performance curves for diverse culvert types.


---

## 2. Simulation


### Interactive Culvert Design Tool – Inlet vs Outlet Control

This tool estimates and visualizes flow through a culvert under both **inlet control** and **outlet control** conditions, following standard hydraulic design principles from FHWA HDS-5.

###  Flow Control Regimes

### 🔵 Inlet Control
- Occurs when the culvert barrel is not flowing full.
- Flow depends on:
  - Entrance geometry (weir or orifice behavior)
  - Headwater elevation \( H_e \)
  - Culvert width or cross-sectional area
- Two options:
  - **Weir flow**: $$
  Q = C_d \cdot b \cdot H_e^{1.5}
  $$
  - **Orifice flow**: $$
  Q = C_d \cdot A \cdot \sqrt{2gH_e}
  $$

### 🟢 Outlet Control
- Occurs when the culvert flows full and tailwater affects the flow.
- Governed by **Manning’s equation**:
  $
  Q = \frac{1}{n} \cdot A \cdot R^{2/3} \cdot \sqrt{S}
  $

- Depends on:
  - Culvert shape and size
  - Hydraulic radius $( R = A / P $)
  - Slope $( S $)
  - Roughness coefficient $( n $)

---

###  How It Works

You control the following parameters:

- **Culvert Shape**: Circular or Box (square)
- **Diameter or Depth** $( d $): Defines area $( A $) and perimeter $( P $)
- **Length $( L $)** and **Slope $( S $)**
- **Headwater depth** $( H_e $) and **Tailwater depth** $( H_c $)
- **Inlet Type**: Weir or Orifice
- **Manning's n**: Roughness (based on material)

For each scenario, the tool computes:

- Discharge under **inlet control**
- Discharge under **outlet control**
- Governing condition based on which flow is more restrictive
- **Visual plot** comparing both flow regimes over a range of heads

---

### Output Highlights

- 💡 **Flow curves** for inlet and outlet control
- 🔵 **Vertical markers** at selected inlet and outlet heads
- ✅ Automatically identifies **governing condition**
- Adaptable for **preliminary design, teaching**, and hydraulic analysis

In [1]:
import numpy as np
import matplotlib.pyplot as plt
from ipywidgets import interact, FloatSlider, Dropdown

g = 9.81  # m/s² gravity

# Inlet Control Flow (weir or orifice equation, depends on head and entrance type)
def culvert_inlet_control(Q_type, A, He, b, Cd_weir=0.6, Cd_orifice=0.5):
    if Q_type == "Weir":
        return Cd_weir * b * He**1.5
    elif Q_type == "Orifice":
        return Cd_orifice * A * np.sqrt(2 * g * He)
    else:
        return 0

# Outlet Control Flow (full barrel, based on Manning's equation)
def culvert_outlet_control(n, A, R, S):
    return (1/n) * A * R**(2/3) * S**0.5

# Wrapper function for visualization and comparison
def culvert_design(d, L, S, n, He, Hc, Q_type, shape):
    A = np.pi * (d / 2)**2 if shape == "Circular" else d * d
    P = np.pi * d if shape == "Circular" else 2 * d + 2 * d
    R = A / P

    # Inlet Control
    Q_inlet = culvert_inlet_control(Q_type, A, He, d)

    # Outlet Control
    Q_outlet = culvert_outlet_control(n, A, R, S)

    print("🧮 Culvert Design Summary")
    print(f"Shape: {shape}, Diameter/Depth: {d:.2f} m, Length: {L:.1f} m, Slope: {S:.4f}")
    print(f"Headwater Elevation (inlet): {He:.2f} m, Tailwater Elevation (outlet): {Hc:.2f} m")
    print(f"🔵 Inlet Control Discharge (Q_in): {Q_inlet:.3f} m³/s")
    print(f"🟢 Outlet Control Discharge (Q_out): {Q_outlet:.3f} m³/s")

    governing = "Inlet Control" if Q_inlet < Q_outlet else "Outlet Control"
    print(f"✅ Governing Flow Control: {governing}")

    # Optional plot
    heads = np.linspace(0.01, max(He, Hc) * 1.5, 200)
    Q_in = [culvert_inlet_control(Q_type, A, H, d) for H in heads]
    Q_out = [culvert_outlet_control(n, A, R, S)] * len(heads)

    plt.figure(figsize=(8, 5))
    plt.plot(heads, Q_in, label="Inlet Control", color="orange")
    plt.plot(heads, Q_out, label="Outlet Control", color="green", linestyle="--")
    plt.axvline(He, color="blue", linestyle=":", label="Inlet Head")
    plt.axvline(Hc, color="red", linestyle=":", label="Tailwater Depth")
    plt.xlabel("Head H (m)")
    plt.ylabel("Discharge Q (m³/s)")
    plt.title("Culvert Flow: Inlet vs Outlet Control")
    plt.grid(True, linestyle="--", alpha=0.6)
    plt.legend()
    plt.tight_layout()
    plt.show()

# Interactive controls
interact(
    culvert_design,
    d=FloatSlider(value=1.0, min=0.3, max=5.0, step=0.1, description="Diameter d (m)"),
    L=FloatSlider(value=10, min=1, max=50, step=1, description="Length L (m)"),
    S=FloatSlider(value=0.01, min=0.0001, max=0.05, step=0.0005, readout_format=".4f", description="Slope S"),
    n=FloatSlider(value=0.013, min=0.010, max=0.035, step=0.001, description="Manning n"),
    He=FloatSlider(value=1.0, min=0.1, max=5.0, step=0.1, description="Inlet Head He"),
    Hc=FloatSlider(value=0.5, min=0.0, max=5.0, step=0.1, description="Tailwater Depth Hc"),
    Q_type=Dropdown(options=["Weir", "Orifice"], value="Weir", description="Inlet Type"),
    shape=Dropdown(options=["Circular", "Box"], value="Circular", description="Culvert Shape")
)

interactive(children=(FloatSlider(value=1.0, description='Diameter d (m)', max=5.0, min=0.3), FloatSlider(valu…

<function __main__.culvert_design(d, L, S, n, He, Hc, Q_type, shape)>

## 3. Self-Assessment

###  Culvert Design Learning Module – Inlet vs Outlet Control

This resource supports the interactive culvert design tool by deepening conceptual understanding and promoting reflective thinking in hydraulic engineering.

---

### Conceptual Questions (with Hints)

1. **What distinguishes inlet control from outlet control in culvert hydraulics, and why does it matter for culvert design?**  
   🔎 *Hint: Think about whether flow is limited by the entrance geometry or by what happens inside the barrel.*

2. **How does Manning’s equation reflect the influence of pipe roughness, geometry, and slope in outlet-controlled culvert flow?**  
   🔎 *Hint: Pay attention to the role of the roughness coefficient \( n \), hydraulic radius \( R \), and slope \( S \).*

3. **Why does the governing control condition depend on the comparison between inlet and outlet discharge capacities?**  
   🔎 *Hint: Consider which component of the culvert system is more restrictive to flow under given conditions.*

4. **When modeling inlet control, why are both weir and orifice equations used? What do they represent physically?**  
   🔎 *Hint: It depends on whether the flow is free over the crest or fully submerged at the entrance.*

5. **Explain how culvert shape (circular vs. box) affects flow area and perimeter, and thus flow regime and velocity.**  
   🔎 *Hint: Different shapes result in different hydraulic radii, affecting resistance and capacity.*

---

### 🔍 Reflective Questions (with Hints)

1. **In mountainous terrain, which control (inlet or outlet) likely governs culvert flow, and why?**  
   🔎 *Hint: Think about high slopes, short culvert lengths, and swift approach flows.*

2. **How might debris accumulation at the entrance affect the accuracy of inlet control predictions?**  
   🔎 *Hint: Imagine how effective flow area and entrance geometry could change in practice.*

3. **If tailwater level increases over time due to downstream changes, what happens to the governing control assumption?**  
   🔎 *Hint: A rising outlet depth could push the system into full-pipe flow conditions.*

4. **Why is it important to account for both frequent low flows and rare flood events in culvert design?**  
   🔎 *Hint: Reliability and safety aren’t based on a single storm. Think resilience and serviceability.*

5. **How might increased rainfall intensity due to climate change affect culvert sizing and design safety margins?**  
   🔎 *Hint: Consider whether current designs are conservative enough for future extremes.*

---

### 🧪 General Problem & Solution

**Problem:**  
A circular culvert has diameter \( d = 1.2 \) m, length \( L = 30 \) m, slope \( S = 0.01 \), and Manning's \( n = 0.013 \). The headwater depth at the entrance is \( H_e = 1.0 \) m, and tailwater depth is \( H_c = 0.5 \) m. Determine the inlet and outlet control discharges for both weir and orifice entrance conditions. Assume:

- $( C_d^{\text{weir}} = 0.6 $)
- $( C_d^{\text{orifice}} = 0.5 $)

**Solution:**

1. Cross-sectional area of circular culvert:
   $$
   A = \pi \left( \frac{1.2}{2} \right)^2 \approx 1.13\ \text{m²}
   $$

2. Wetted perimeter:
   $$
   P = \pi \cdot 1.2 \approx 3.77\ \text{m}
   $$

3. Hydraulic radius:
   $$
   R = \frac{A}{P} \approx \frac{1.13}{3.77} \approx 0.30\ \text{m}
   $$

4. Inlet Control (Weir Flow):
   $$
   Q_{\text{in,weir}} = 0.6 \cdot 1.2 \cdot 1.0^{1.5} = 0.72\ \text{m³/s}
   $$

5. Inlet Control (Orifice Flow):
   $$
   Q_{\text{in,orifice}} = 0.5 \cdot 1.13 \cdot \sqrt{2 \cdot 9.81 \cdot 1.0} \approx 2.50\ \text{m³/s}
   $$

6. Outlet Control:
   $$
   Q_{\text{out}} = \frac{1}{0.013} \cdot 1.13 \cdot 0.30^{2/3} \cdot \sqrt{0.01} \approx 1.80\ \text{m³/s}
   $$

**Conclusion:** The governing discharge is the lower of the estimates:  
✅ Governing flow = **0.72 m³/s** (inlet control, weir condition)

---

### ✅ Quiz Questions

**Q1.** Inlet control occurs when:  
A. The barrel is flowing full  
B. Tailwater controls capacity  
C. Entrance geometry and headwater dominate  
D. The outlet is submerged  
🟢 **Correct Answer:** C

---

**Q2.** Which of the following increases outlet-controlled flow?  
A. Higher Manning’s \( n \)  
B. Steeper culvert slope  
C. Longer culvert  
D. Smaller cross-sectional area  
🟢 **Correct Answer:** B

---

**Q3.** The flow area of a circular culvert with diameter 1.0 m is approximately:  
A. 0.79 m²  
B. 1.00 m²  
C. 3.14 m²  
D. 0.50 m²  
🟢 **Correct Answer:** A

---

**Q4.** Which is true about weir and orifice inlet equations?  
A. Weir equation is more valid when culvert is submerged  
B. Orifice equation is used for shallow flow  
C. Weir flow occurs when water just begins overtopping  
D. Orifice flow assumes zero entrance loss  
🟢 **Correct Answer:** C

---

**Q5.** If a culvert design shows $( Q_{\text{in}} > Q_{\text{out}} $), which condition governs?  
A. Inlet Control  
B. Outlet Control  
C. Submerged Flow  
D. Critical Flow  
🟢 **Correct Answer:** B